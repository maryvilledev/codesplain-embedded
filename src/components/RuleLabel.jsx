import React, { PropTypes } from 'react';

const styles = {
  countSpan: {
    color: '#666666',
    display: 'block',
    float: 'right',
  },
  label: {
    borderRadius: '5px',
    borderStyle: 'solid',
    borderWidth: '1px',
    cursor: 'pointer',
    display: 'inline-block',
    fontSize: '10pt',
    lineHeight: 1,
    marginTop: '5px',
    marginRight: '5px',
    maxWidth: '200px',
    padding: '5px',
    width: '100%',
  },
};

const RuleLabel = ({ color, rule, count }) => {
  const backgroundColor = color;
  const borderColor = 'transparent';

  return (
    <div style={{ ...styles.label, backgroundColor, borderColor }} >
      {rule}
      <span style={styles.countSpan}>{`(${count})`}</span>
    </div>
  );
};

RuleLabel.propTypes = {
  color: PropTypes.string.isRequired,
  count: PropTypes.number.isRequired,
  rule: PropTypes.string.isRequired,
};

export default RuleLabel;
