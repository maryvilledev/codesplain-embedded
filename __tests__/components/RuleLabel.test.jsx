import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

import RuleLabel from '../../src/components/RuleLabel';

describe('<RuleLabel />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <RuleLabel
        color="blue"
        count={1}
        rule="None"
        selected={true}
        value="nada"
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
