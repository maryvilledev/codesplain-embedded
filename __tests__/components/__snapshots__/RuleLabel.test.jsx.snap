// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RuleLabel /> matches snapshot 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "blue",
      "borderColor": "transparent",
      "borderRadius": "5px",
      "borderStyle": "solid",
      "borderWidth": "1px",
      "cursor": "pointer",
      "display": "inline-block",
      "fontSize": "10pt",
      "lineHeight": 1,
      "marginRight": "5px",
      "marginTop": "5px",
      "maxWidth": "200px",
      "padding": "5px",
      "width": "100%",
    }
  }
>
  None
  <span
    style={
      Object {
        "color": "#666666",
        "display": "block",
        "float": "right",
      }
    }
  >
    (1)
  </span>
</div>
`;
